pkg/lib/sidecars/pods/filter.go

```diff
@@ -64,6 +66,11 @@ func isPodRunning(pod v1.Pod) bool {
 	return pod.Status.Phase == v1.PodRunning
 }
 
+func hasCustomImageAnnotation(pod v1.Pod) bool {
+	_, found := pod.Annotations[istioSidecarCustomImageAnnotation]
```

Comment:
Won't this fail if `pod.Annotations` will be empty/nil?

Reply:
It will not fail in such case. Instead the `found`  variable will be populated with false.